theano/gof/opt.py

```diff
@@ -1353,21 +1353,27 @@ def transform(self, node):
                     self.process_count[opt] += 1
                 if not new_repl:
                     continue
-                else:
-                    if self.profile:
-                        self.node_created[opt] += len(graph.ops(fgraph.variables, new_repl))
-                        self.applied_true[opt] += 1
-                    break  # break from the for loop over optimization.
+                if isinstance(new_repl, (tuple, list)):
+                    new_vars = new_repl
+                else:  # It must be a dict
+                    new_vars = list(new_repl.values())
```

Comment:
We don't care of the order of the dict. This is just to count the new variable introduced in the graph. We don't use it to change the graph. Line 1371 return new_repl